lib/index.js

```diff
@@ -74,20 +108,39 @@ function OniyiHttpClient(params = {}) {
     );
 
     // augment requestParams to properly handle absolute uris
-    requestParams = parseUri(requestParams);
-
-    // call request directly, if there are no plugins loaded in this client
-    if (!Array.isArray(plugins) || plugins.length < 1) {
-      logger.debug('no plugins loaded, uri: %s', initialParams.uri);
-      return new request.Request(requestParams);
-    }
+    requestOptions = parseUri(requestOptions);
+
+    const ctx = { requestOptions, hookState };
+    // save reference to the original callback
+    const { callback } = requestOptions;
+    // https://github.com/kriskowal/q#using-deferreds
+    const { promise, resolve, reject } = q.defer();
```

Comment:
This is where I've used `q`. IMHO it makes code more readable then regular `return new Promise` syntax.
q.[defer](https://github.com/kriskowal/q#using-deferreds)()

Reply:
in this case "less dependencies" wins over subjective readability of code.

```js
const promise = new Promise((resolve, reject) =>
      requestPhaseList.run(ctx, phaseError => {
        if (phaseError) {
          return _.isFunction(callback)
            ? callback(phaseError)
            : reject(phaseError)
        }
        displayPluginsLoadingData(hookState)

        if (_.isFunction(callback)) {
          // override the current callback in order to add new phase list
          // responsible for handling/manipulating a response object
          requestOptions.callback = (err, response, body) => {
            // TODO: add a response phaseList

            callback(err, response, body)
          }
        }

        return resolve(new request.Request(requestOptions)) // eslint-disable-line no-new
      }),
    );

    return promise;
```